### PR TITLE
Enable golint and fix errors

### PR DESCRIPTION
**Pull request by gssbzn**
This enables golint to be run as part of make lint and fixes some warnings flagged by it
this aims to make the codebase more homogenous as part of the code seemed to be linted and other not

**Warning**: this may be a breaking change as the name of some public interfaces are changing

### CODE REVIEW

File: mongodbatlas/clusters.go [hunk 1]
```diff
@@ -114,61 +114,6 @@ type clustersResponse struct {
 	TotalCount int       `json:"totalCount,omitempty"`
 }
 
-// DefaultDiskSizeGB represents the Tier and the default disk size for each one
-// it can be use like: DefaultDiskSizeGB["AWS"]["M10"]
-var DefaultDiskSizeGB map[string]map[string]float64 = map[string]map[string]float64{
```
Comment (gssbzn) [inline mongodbatlas/clusters.go]: this was not in use internally and there's no way for you to use it as `mongodbatlas. DefaultDiskSizeGB` so there's no point on keeping something like this
This is the easy approach but I'm happy to revert this or expose it as a `func`

**Comment by themantissa** [comment]
Note: reaching out internally to @gssbzn.  Not assigning reviewers until then.

File: mongodbatlas/clusters.go (hunk 1 above)
Comment (PacoDw) [reply mongodbatlas/clusters.go]: Could you remove it again please? it isn't already necessary in the Terraform part.

File: mongodbatlas/clusters.go [hunk 2]
```diff
@@ -114,6 +114,60 @@ type clustersResponse struct {
 	TotalCount int       `json:"totalCount,omitempty"`
 }
 
+// DefaultDiskSizeGB represents the Tier and the default disk size for each one
+// it can be use like: DefaultDiskSizeGB["AWS"]["M10"]
+var DefaultDiskSizeGB map[string]map[string]float64 = map[string]map[string]float64{
```
Comment (PacoDw) [inline mongodbatlas/clusters.go]: Could you remove it again please? it isn't already necessary in the Terraform part.

Comment (gssbzn) [reply mongodbatlas/clusters.go]: Done

Comment (PacoDw) [reply mongodbatlas/clusters.go]: Thank you so much 👍

**Pull request merged**